src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/indentation/indentation/InputIndentationRecordPattern.java

```diff
@@ -0,0 +1,100 @@
+ * forceStrictCondition = true                                              //indent:1 exp:1
+ */                                                                         //indent:1 exp:1
+public class InputIndentationRecordPattern {  //indent:0 exp:0
+    record ColoredPoint(boolean p, int x, int c) { } //indent:4 exp:4
+    record Rectangle(ColoredPoint upperLeft, ColoredPoint lowerRight) { } //indent:4 exp:4
+
+    void test(Object obj) { //indent:4 exp:4
+        boolean b = obj instanceof //indent:8 exp:8
+                ColoredPoint(_,_,_);  //indent:16 exp:12 warn
+        boolean c = obj instanceof //indent:8 exp:8
+            ColoredPoint(_,_,_);  //indent:12 exp:12
+
+        boolean bb = obj instanceof //indent:8 exp:8
+        ColoredPoint(_,_,_);  //indent:8 exp:12 warn
+
+        if (obj instanceof ColoredPoint(_,_,_)) {} //indent:8 exp:8
+
+        if (obj instanceof //indent:8 exp:8
+                ColoredPoint(_,_,_)) {} //indent:16 exp:12 warn
+        if (obj instanceof     //indent:8 exp:8
+            ColoredPoint(_,_,_)) {} //indent:12 exp:12
+
+        if (obj instanceof  //indent:8 exp:8
+        ColoredPoint(_,_,_)) {  } //indent:8 exp:12 warn
+
+        if (obj instanceof //indent:8 exp:8
+ColoredPoint(_,_,_)) { }  //indent:0 exp:12 warn
+        if (obj instanceof //indent:8 exp:8
+ColoredPoint( //indent:0 exp:12 warn
+_,  //indent:0 exp:12 warn
+_,  //indent:0 exp:12 warn
+_)) {} //indent:0 exp:12 warn
+
+        if (obj instanceof //indent:8 exp:8
+            ColoredPoint( //indent:12 exp:12
+            _,  //indent:12 exp:12
+            _,  //indent:12 exp:12
+            _)) {} //indent:12 exp:12
+
+        if (obj instanceof ColoredPoint( //indent:8 exp:8
+            _,  //indent:12 exp:12
+            _,  //indent:12 exp:12
+            _)) {} //indent:12 exp:12
+
+        if (obj instanceof //indent:8 exp:8
+                Rectangle( //indent:16 exp:12 warn
+                ColoredPoint(  //indent:16 exp:12 warn
+                        boolean x, //indent:24 exp:12 warn
+                int y, //indent:16 exp:12 warn
+                        _), //indent:24 exp:12 warn
+                ColoredPoint(_,_,_)  //indent:16 exp:12 warn
+                ) //indent:16 exp:8 warn
+        ) {} //indent:8 exp:8
+
+        if (obj instanceof //indent:8 exp:8
+Rectangle(   //indent:0 exp:12 warn
+ColoredPoint(   //indent:0 exp:12 warn
+        boolean x,  //indent:8 exp:12 warn
+        int y,  //indent:8 exp:12 warn
+        _),  //indent:8 exp:12 warn
+ColoredPoint(_,_,_)  //indent:0 exp:12 warn
+        )) {}   //indent:8 exp:12 warn
+
+        if (obj instanceof //indent:8 exp:8
+            Rectangle(   //indent:12 exp:12
+            ColoredPoint(   //indent:12 exp:12
+            boolean x,  //indent:12 exp:12
+            int y,  //indent:12 exp:12
+            _),  //indent:12 exp:12
+            ColoredPoint(_,_,_)  //indent:12 exp:12
+            )) {}   //indent:12 exp:8 warn
```

Comment:
same here for the record pattern components  named or unnamed. 

I think there is a conflict here  ( 8 -> 12) then (12 -> 8)

```
ColoredPoint(_,_,_)  //indent:0 exp:12 warn
        )) {} 
```
        
```
        ColoredPoint(_,_,_)  //indent:12 exp:12
            )) {}   
```

Reply:
This needs to be a new issue if it can never be satisfied.

Reply:
After looking carefully for this example.I think the problem is we have two parens on this line.  one paren for the `if` and one for the record pattern.

The record pattern paren has to be on indent 12 and the if paren has to be on 8. so putting them on the same level will always produce a violation on one of them.

Sorry, there is no conflict here the check just requires a different indentation for each paren.

Reply:
If you can never satisfy this so there is no violations then this **IS** an issue. All violations should be valid and should be solvable to some extent. Indentation is to guide users to correct indentation. It can't be correct if it is always wrong.

Reply:
I just thought that if this conflict has a valid reason then this is not an issue.but yes users still face a violation that can't be solved
https://github.com/checkstyle/checkstyle/issues/15193